packages/@tinacms/graphql/src/resolver/index.ts

```diff
@@ -511,23 +510,42 @@ export class Resolver {
     }
   }
 
+  public getDocumentsForCollection = async (collectionName: string) => {
+    const collection = this.tinaSchema.getCollection(collectionName)
+    return this.database.store.glob(collection.path, this.getDocument)
+  }
+
   public resolveCollectionConnection = async ({
     args,
     lookup,
   }: {
     args: Record<string, Record<string, object>>
     lookup: CollectionDocumentListLookup
   }) => {
-    const documents = await this.database.getDocumentsForCollection(
-      lookup.collection
-    )
+    let documents
+    if (args.filter) {
+      const queries = []
+      const flattenedArgs = flat(args.filter, { delimiter: '#' })
+      Object.entries(flattenedArgs).map(([key, value]) => {
+        const keys = key.split('#')
+        const realKey = keys.slice(0, keys.length - 1).join('#')
+        queries.push(
+          `__attribute__${lookup.collection}#${lookup.collection}#${realKey}#${value}`
```

Comment:
@jeffsee55 From our chat, this should be `${lookup.collection}#${lookup.template}` - not double `collection`.